pkg/distributor/distributor.go

```diff
@@ -280,14 +281,55 @@ func (d *Distributor) checkSample(ctx context.Context, userID, cluster, replica
 	return true, nil
 }
 
+// Validates a single series from a write request. Will remove HA labels if necessary.
```

Comment:
State what the return values are.
Include why we have return values both via pointer and by value.

Reply:
Done. I've simplified the function as well so the pointer isn't needed anymore.